ZenPacks/zenoss/Import4/bin/dispatch.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+#
+# This script starts the processing of rrdfiles by placing
+# chunks of tasks (files containing list of rrdfile names)
+# into a task directory monitored by the converter workers.
+#
+# This script is to be run once per rrd import for the source rrd tree
+#
+export rrdtop="$1"          # top directory (in absolute path) of the source rrd tree
+export tasktop="$2"         # task queue directory for the rrd file lists (also in absolute path)
+
+# derived
+export tasktmp="$2/.tmp"
+
+# common block
+progdir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+. "$progdir/utils.sh"
+
+# finds all the rrd files and group them into task files
+# The result is placed into the provied task queue directory
+# The target directory is cleaned first
+
+if [[ ! -d "$rrdtop" ]] 
+then
+    echo "Usage: $0 <rrd_dir> <task_dir>" >&2
+    err_exit "rrd directory '$rrdtop' does not exist ..." 
+fi
+
+# prep the target dirs
+mkdir -p "$tasktmp"                                                     || err_exit "Cannot create $tasktop and $tasktmp"
+
+# clean up the target directory
+rm -rf "$tasktop"/task.*                                                     || err_exit "Cannot remove files under $tasktop" 
+rm -rf "$tasktmp"                                                       || err_exit "Cannot remove $tasktmp"
+
+# finish preparing the tasks first before moving them to the final dir
+# Assertion:
+# if a task file exists under tasktop, no more change will be made to it
+mkdir -p "$tasktmp"                                                     || err_exit "Cannot create $tasktop or $tasktmp"
+find "$rrdtop" -type f -name "*.rrd" | split -l 10 - "$tasktmp"/task.   || err_exit "Error generating tasks"
```

Comment:
Mainly just a knob for progress report. (10~30 seconds per batch)
It may have impact on sort | uniq operation if left too big.
It may spend too much book keeping time if left to small.
The end user may not actually make much use of the number.
So, 10, is just a empirical sweet spot number.